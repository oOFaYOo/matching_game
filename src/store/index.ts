import {configureStore} from '@reduxjs/toolkit'
import MatchingGameReducer from './slice'

export const store = configureStore({
    reducer: {
        MatchingGameStore: MatchingGameReducer,
    },
})

export type RootState = ReturnType<typeof store.getState>