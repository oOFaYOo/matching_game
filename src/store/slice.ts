import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'
import {ICard, IMatchingGameState} from "../types";

export const initialState: IMatchingGameState = {
    theme: localStorage.theme ? localStorage.theme : 'light',
    topic: localStorage.topic ? localStorage.topic : '',
    amount: localStorage.amount ? localStorage.amount : '',
};

export const MatchingGameStoreSlice = createSlice({
    name: 'MatchingGameStore',
    initialState,
    reducers: {
        setTheme : (state, action: PayloadAction<'light' | 'dark'>) => {
            state.theme = action.payload
        },
        setTopic : (state, action: PayloadAction<string>) => {
            state.topic = action.payload
        },
        setAmount : (state, action: PayloadAction<string>) => {
            state.amount = action.payload
        },
    },
});

export const {
    setTheme,
    setTopic,
    setAmount,
} = MatchingGameStoreSlice.actions;

export default MatchingGameStoreSlice.reducer;