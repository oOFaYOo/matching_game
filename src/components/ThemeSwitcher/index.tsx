import React from "react";
import {useDispatch, useSelector} from "react-redux";
import {Switch} from "@mui/material";
import {setTheme} from "../../store/slice";
import {RootState} from "../../store";
import sun from './../../images/sun.png'
import moon from './../../images/moon.png'

const ThemeSwitcher = () => {

    const {theme} = useSelector((state: RootState) => state.MatchingGameStore);
    const dispatch = useDispatch();

    return (
        <div className={'flex flex-row items-center'}>
            <img src={sun} className={`${theme === 'dark' ? 'grayscale' : ''} w-28 h-28`}/>
            <Switch data-testid={'switch'} id={'theme-switcher'} color={'default'} checked={theme !== 'light'}
                    onChange={(e) => {
                        const theme = e.currentTarget.checked ? 'dark' : 'light';
                        localStorage.setItem('theme', theme)
                        dispatch(setTheme(theme));
                    }}/>
            <img src={moon} className={`${theme === 'dark' ? '' : 'grayscale'} w-28 h-28`}/>
        </div>
    )
}

export default ThemeSwitcher;