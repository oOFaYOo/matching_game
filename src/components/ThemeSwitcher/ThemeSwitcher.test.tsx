import React from "react";
import {fireEvent, render, screen} from "@testing-library/react";
import {Provider} from "react-redux";
import {store} from "../../store";
import ThemeSwitcher from "./index";

jest.mock('./../../images/sun.png', ()=>'');
jest.mock('./../../images/moon.png', ()=>'');


it('ThemeSwitcher test', () => {

    const resizeWindow = (x:number, y:number) => {
        window.innerWidth = x;
        window.innerHeight = y;
        window.dispatchEvent(new Event('resize'));
    }

    resizeWindow(500, 300);

    const Comp = () => {
        return (
            <Provider store={store}>
                <ThemeSwitcher />
            </Provider>
        )
    }

    const {container} = render(<Comp/>);
    fireEvent.click( screen.getByTestId("switch").childNodes[0]);

    fireEvent.click(container.getElementsByClassName('mobile:w-48 mobile:h-48 tablet:w-36 tablet:h-36')[0]);
    // expect(container).toMatchSnapshot();
})