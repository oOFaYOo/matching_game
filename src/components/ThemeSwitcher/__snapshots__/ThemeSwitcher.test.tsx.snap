// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitcher test 1`] = `
<div>
  <div
    class="big:flex flex-row items-center mobile:hidden tablet:hidden"
  >
    <img
      class=" w-20 h-20"
      src=""
    />
    <span
      class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
    >
      <span
        class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorDefault PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorDefault css-1i8lvs-MuiButtonBase-root-MuiSwitch-switchBase"
        data-testid="switch"
      >
        <input
          class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
          id="theme-switcher"
          type="checkbox"
        />
        <span
          class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
      />
    </span>
    <img
      class="grayscale w-20 h-20"
      src=""
    />
  </div>
  <div
    class="flex flex-row items-center big:hidden block"
  >
    <img
      class="mobile:w-48 mobile:h-48 tablet:w-36 tablet:h-36"
      src=""
    />
  </div>
</div>
`;
